### PR TITLE
Adds option to disable default @vite/pugin-react injection

### LINKED ISSUE #435: Disable default react refresh through ladle config
**Is your feature request related to a problem? Please describe.**
Most projects in the react ecosystem are very easy to use with Preact. This is true for ladle as well, although a strange pattern is introduced when trying to preserve fast-refresh functionality.

Making ladle work with preact is as simple as including the preact preset in a custom vite config. The problem though is that you have to use the `prefreshEnabled: false` option, otherwise an error is through in the browser console:` debug.js:80 Uncaught SyntaxError: Identifier 'prevRefreshReg' has already been declared`

Here's an example of the working config for no HMR:
```js
// vite.config.js
import preact from '@preact/preset-vite';

export default defineConfig({
  plugins: [
    preact({ prefreshEnabled: false }),
  ],
})
```
Presumably, this is because both `@vitejs/plugin-react` and `@preact/preset-vite` are injecting their own versions of `prevRefreshReg`.

So if you want fast-refresh with preact, you have to do something which I consider very peculiar. You have to install `@vitejs/plugin-react` for the sole purpose of importing it into your vite config and using it to exclude all files.  From my testing, this was the only way to effectively prevent the double injections. So here's an example of that config:

```js
// vite.config.js
import react from "@vitejs/plugin-react";

export default defineConfig({
  plugins: [
    react({
      exclude: [
        '*',
        '**/*',
        '**/**/*',
      ]
    }),
    preact({ prefreshEnabled: true }),
  ],
})
```

**Describe the solution you'd like**
To me, it makes much more sense to provide a config flag that prevents ladle from injecting the `@vitejs/plugin-react` in the first place. It's a strange pattern to need to install a dependency in order to turn it off the functionality that gets auto-injected by the same (sub)dependency. 

Ideally, something in the ladle config would bypass the check for the react plugin in the vite config. This would let users define their own configs without needing to provide valid instances of the react plugin inside the vite config.

**Describe alternatives you've considered**
The only alternatives I see are losing fast-refresh for preact projects or using the strange pattern described above.

**Additional context**

### PR DESCRIPTION
**Pull request by Scott-Fischer**
This PR adds a new ladle config option to support cases where the user does not desire the automatic injection of  `@vite/pugin-react` into their vite config. 

A good example of this is to support fast-refresh functionality in preact projects (as documented in issue  #435).

After this PR lands in main, I'd be happy to add a preact recipe to the docs in a followup PR.
Closes #435

### CODE REVIEW

File: packages/ladle/lib/cli/vite-base.js [hunk 1]
```diff
@@ -60,6 +60,10 @@ const getBaseViteConfig = async (ladleConfig, configFolder, viteConfig) => {
   debug("User provided @vite/plugin-react: %s", hasReactPlugin);
   debug("User provided @vite/plugin-react-swc: %s", hasReactSwcPlugin);
 
+  const hasDisabledReactPlugin = ladleConfig?.disableReactPlugin;
```
Comment (tajo) [inline packages/ladle/lib/cli/vite-base.js:63]: are we missing the part where we use this setting to actually disable the plugin?

Comment (justb3a) [reply packages/ladle/lib/cli/vite-base.js:63]: @Scott-Fischer ? I would love to see this PR finished 🙏

**Comment by jrson83** [comment]
@Scott-Fischer any news on this? Would be great if you add the requested changes.

**Comment by Scott-Fischer** [comment]
@jrson83 @justb3a Sorry for the delay on this. I got distracted with another project. I'm hoping to finish this PR at some point this week.

**Comment by jrson83** [comment]
@tajo anything holding back merge? Would be great to have it soon.

**Comment by tajo** [comment]
> @tajo anything holding back merge? Would be great to have it soon.

I don't think the implementation works right now? `hasDisabledReactPlugin` is not really being used

File: packages/ladle/lib/cli/vite-base.js (hunk 1 above)
Comment (Scott-Fischer) [reply packages/ladle/lib/cli/vite-base.js:63]: @tajo Yeah it was missing! 🤦 

Thanks for the catch.

**Comment by Scott-Fischer** [comment]
@tajo I seem to have potentially royally borked up this PR by trying to pull in the upstream changes to my fork. Any idea why it's showing so many file changes in PR comparison but only a few in my fork? https://github.com/tajo/ladle/compare/main...Scott-Fischer:ladle:main

**Comment by tajo** [comment]
Have you rebased your main branch? Anyway, you can always close this and create a new PR

**Pull request closed without merging**